### PR TITLE
ui fixes

### PR DESCRIPTION
**Pull request by dinportnoy**

### CODE REVIEW

File: quix-frontend/client/src/react-components/file-explorer/TreeItemMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  setIsOpen,
+  onMenuClick,
+}: TreeItemMenuProps) => {
+  const iconRef = useRef(null);
+  const menuWrapperRef = useRef(null);
+  useOutsideAlerter([iconRef, menuWrapperRef], () => setIsOpen(false));
+
+  return (
+    <Dropdown
+      OptionsWrapper={({options}: any) =>
```
Comment (antonpodolsky) [inline quix-frontend/client/src/react-components/file-explorer/TreeItemMenu.tsx:27]: why not `optionsWrapper`?

File: quix-frontend/client/src/react-components/file-explorer/TreeItemMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        <ul
+          ref={menuWrapperRef}
+          className="bi-dropdown-menu bi-tree-item-dropdown bi-fade-in"
+        >
+          {options}
+        </ul>
+      }
+      ReferenceElement={
```
Comment (antonpodolsky) [inline quix-frontend/client/src/react-components/file-explorer/TreeItemMenu.tsx:35]: why not `referenceElement`?
also, what does reference element mean?

**Review by antonpodolsky — CHANGES_REQUESTED** [review]

File: quix-frontend/client/src/services/hooks/use-outside-alerter.ts [hunk 3]
```diff
@@ -2,28 +2,27 @@ import { useEffect } from 'react';
...
-      document.addEventListener("mousedown", handleClickOutside);
-      return () => {
-          // Unbind the event listener on clean up
-          document.removeEventListener("mousedown", handleClickOutside);
-      };
+    document.addEventListener("mousedown", handler);
+    return () => {
+      document.removeEventListener("mousedown", handler);
+    };
   }, [refs]);
```
Comment (antonpodolsky) [inline quix-frontend/client/src/services/hooks/use-outside-alerter.ts:27]: what happens when the refs change during the component lifecycle?
who's make sure to remove the listeners from the document?

**Review by antonpodolsky — CHANGES_REQUESTED** [review]

File: quix-frontend/client/src/services/hooks/use-outside-alerter.ts (hunk 3 above)
Comment (antonpodolsky) [reply quix-frontend/client/src/services/hooks/use-outside-alerter.ts:27]: the only time listeners are removed is when the component is destroyed.
this whole setup feels like a big memory leak?

Comment (dinportnoy) [reply quix-frontend/client/src/services/hooks/use-outside-alerter.ts:27]: This function wrapped with `useEffect`, which means that every time that a ref changed, the cleanup run

File: quix-frontend/client/src/react-components/file-explorer/TreeItemMenu.tsx (hunk 1 above)
Comment (dinportnoy) [reply quix-frontend/client/src/react-components/file-explorer/TreeItemMenu.tsx:27]: OptionsWrapper's type is `React.ComponentType<InjectedDropdownProps>`
which means it used like:
`<OptionsWrapper .... />`

File: quix-frontend/client/src/react-components/file-explorer/TreeItemMenu.tsx (hunk 2 above)
Comment (dinportnoy) [reply quix-frontend/client/src/react-components/file-explorer/TreeItemMenu.tsx:35]: changed to `element`

File: quix-frontend/client/src/lib/ui/components/dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -1,53 +1,88 @@
...
-  spanClass,
+  classes,
   placement,
+  children,
+  states,
 }: DropdownProps) => {
-  const [_referenceElement, setReferenceElement] = useState(null);
-  const [referenceOptions, setReferenceOptions] = useState(null);
-  const { styles, attributes } = usePopper(_referenceElement, referenceOptions, {
+  createDropdownHelperIfNotExist();
```
Comment (antonpodolsky) [inline quix-frontend/client/src/lib/ui/components/dropdown/Dropdown.tsx:42]: why is this needed?

File: quix-frontend/client/src/lib/ui/components/dropdown/Dropdown.tsx [hunk 5]
```diff
@@ -1,53 +1,88 @@
...
+  const isOpenDefined = !isNil(isOpen);
+
+  const [_isOpen, setIsOpen] = useState(false);
+  const referenceToggle = useRef(null);
+  const referenceOptions = useRef(null);
+
+  useOutsideAlerter([referenceToggle, referenceOptions], () => setIsOpen(false));
```
Comment (antonpodolsky) [inline quix-frontend/client/src/lib/ui/components/dropdown/Dropdown.tsx]: lets give this hook a more meaningful name, right now it doesn't sound like anything related to dom element clicks.

File: quix-frontend/client/src/lib/ui/components/dropdown/Dropdown.tsx [hunk 6]
```diff
@@ -1,53 +1,88 @@
...
+      >
+        {toggle}
       </span>
-      {isOpen
-        ? ReactDOM.createPortal(
-            <div
-              style={styles.popper}
-              {...attributes.popper}
-              ref={setReferenceOptions as any}
+        {ReactDOM.createPortal(
```
Comment (antonpodolsky) [inline quix-frontend/client/src/lib/ui/components/dropdown/Dropdown.tsx]: why do we need to use portal here? doesn't popper.js handle element positioning?

**Review by antonpodolsky — CHANGES_REQUESTED** [review]
